parsec/interfaces/dtd/insert_function.c

```diff
@@ -736,6 +736,28 @@ parsec_dtd_add_profiling_info_generic(parsec_taskpool_t *tp,
     free(str);
 }
 
+void *parsec_dtd_task_profile_info(void *dst, const void *task_, size_t size)
```

Comment:
This function looks significantly more expensive than what we have in today. If we put the info only once on the profiling then the overhead would be small, and we can move forward with this approach. However looking the the code in the parsec core, it seems we are using the TRACE_TASK macro, which dumps the prof_info once per call (so possibly multiple times for each task, increasing the profiling overhead).

Reply:
8f72eda8841da75e788613ffa5ac411e1966d0aa makes sure that the profiling_info function is only called once per task. It also removes explicit calls in DTD and map_operator to this tracing, as tracing of task execution should only be done via PINS.